src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -119,6 +119,33 @@ internal static unsafe string UnicodeToStringFixedArray(UInt16* buffer, int leng
             return result;
         }
 
+        internal static unsafe char* StringToUnicodeBuffer(String str)
+        {
+            if (str == null)
+                return null;
+
+            int stringLength = str.Length;
+
+            if (stringLength == 0)
```

Comment:
This is not correct. Empty string is not a null string. It should be a buffer '\0'.